File: 150.py
class Solution:
    def evalRPN(self, tokens: list[str]) -> int:
        s = []
        operators = {"+", "-", "*", "/"}
        for token in tokens:
            if token not in operators:
                s.append(int(token))
            else:
                arg1 = s.pop()
                arg2 = s.pop()
                if token == "+":
                    s.append(arg2 + arg1)
                elif token == "-":
                    s.append(arg2 - arg1)
                elif token == "/":
                    s.append(int(arg2 / arg1))
                else:
                    s.append(arg2 * arg1)
        return s[0]


print(
    Solution().evalRPN(
        ["10", "6", "9", "3", "+", "-11", "*", "/", "*", "17", "+", "5", "+"]
    )
)

File: 150.cpp
#include <set>
#include <stack>
#include <string>
#include <vector>

using namespace std;

class Solution {
public:
  int evalRPN(vector<string> &tokens) {
    stack<int> s;
    for (auto &token : tokens) {
      if (token == "+") {
        int arg1 = s.top();
        s.pop();
        int arg2 = s.top();
        s.pop();
        s.push(arg2 + arg1);
      } else if (token == "-") {
        int arg1 = s.top();
        s.pop();
        int arg2 = s.top();
        s.pop();
        s.push(arg2 - arg1);
      } else if (token == "*") {
        int arg1 = s.top();
        s.pop();
        int arg2 = s.top();
        s.pop();
        s.push(arg1 * arg2);
      } else if (token == "/") {
        int arg1 = s.top();
        s.pop();
        int arg2 = s.top();
        s.pop();
        s.push(arg2 / arg1);
      } else {
        s.push(stoi(token));
      }
    }
    return s.top();
  }
};